pravega-cli/src/test/java/io/pravega/tools/pravegacli/SecureControllerCommandsTest.java

```diff
@@ -0,0 +1,159 @@
+
+    @Before
+    public void setUp() throws Exception {
+        // Create the secure pravega server to test commands against.
+        LocalPravegaEmulator.LocalPravegaEmulatorBuilder emulatorBuilder = LocalPravegaEmulator.builder()
+                .controllerPort(io.pravega.test.common.TestUtils.getAvailableListenPort())
+                .segmentStorePort(io.pravega.test.common.TestUtils.getAvailableListenPort())
+                .zkPort(io.pravega.test.common.TestUtils.getAvailableListenPort())
+                .restServerPort(io.pravega.test.common.TestUtils.getAvailableListenPort())
+                .enableRestServer(REST_ENABLED)
+                .enableAuth(AUTH_ENABLED)
+                .enableTls(TLS_ENABLED)
+                .restServerPort(REST_SERVER_PORT);
+
+        // Since the server is being built right here, avoiding delegating these conditions to subclasses via factory
+        // methods. This is so that it is easy to see the difference in server configs all in one place. This is also
+        // unlike the ClientConfig preparation which is being delegated to factory methods to make their preparation
+        // explicit in the respective test classes.
+
+        if (AUTH_ENABLED) {
+            emulatorBuilder.passwdFile("src/test/resources/passwd")
+                    .userName(SecurityConfigDefaults.AUTH_ADMIN_USERNAME)
+                    .passwd(SecurityConfigDefaults.AUTH_ADMIN_PASSWORD);
+        }
+        if (TLS_ENABLED) {
+            emulatorBuilder.certFile("src/test/resources/server-cert.crt")
+                    .keyFile("src/test/resources/server-key.key")
+                    .jksKeyFile("src/test/resources/server.keystore.jks")
+                    .jksTrustFile("src/test/resources/client.truststore.jks")
+                    .keyPasswordFile("src/test/resources/server.keystore.jks.passwd");
+        }
+
+        localPravega = emulatorBuilder.build();
+
+        // Set the CLI properties.
+        STATE.set(new AdminCommandState());
+        Properties pravegaProperties = new Properties();
+        pravegaProperties.setProperty("cli.controllerRestUri", "localhost:" + REST_SERVER_PORT.toString());
+        pravegaProperties.setProperty("pravegaservice.zkURL", localPravega.getInProcPravegaCluster().getZkUrl());
+        pravegaProperties.setProperty("pravegaservice.containerCount", "4");
+        pravegaProperties.setProperty("cli.authEnabled", "true");
+        pravegaProperties.setProperty("cli.tlsEnabled", "true");
+        pravegaProperties.setProperty("cli.userName", "admin");
+        pravegaProperties.setProperty("cli.password", "1111_aaaa");
+        pravegaProperties.setProperty("cli.security.tls.trustStore.location", "src/test/resources/client.truststore.jks");
+        STATE.get().getConfigBuilder().include(pravegaProperties);
+
+        localPravega.start();
+        // Wait for the server to complete start-up.
```

Comment:
nit: It's a common practice to have a single-line comment preceded by a blank line.

Reply:
Have made sure to address the same across all changes.